discovery/seata-discovery-kubernetes/src/main/java/io/seata/discovery/registry/kubernetes/KubernetesRegistryServiceImpl.java

```diff
@@ -0,0 +1,183 @@
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.seata.discovery.registry.kubernetes;
+
+import io.fabric8.kubernetes.api.model.EndpointsList;
+import io.fabric8.kubernetes.api.model.Endpoints;
+import io.fabric8.kubernetes.api.model.EndpointSubset;
+import io.fabric8.kubernetes.api.model.EndpointAddress;
+import io.fabric8.kubernetes.api.model.EndpointPort;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.seata.config.Configuration;
+import io.seata.config.ConfigurationFactory;
+import io.seata.discovery.registry.RegistryService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetSocketAddress;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * @author @hero_zhanghao
+ * @date 2019/5/6 17:55
+ **/
+public class KubernetesRegistryServiceImpl  implements RegistryService<KubernetesEventListener> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(KubernetesRegistryServiceImpl.class);
+
+
+    private static final String DEFAULT_NAMESPACE= "any";
+
+    private static final String FILE_ROOT_REGISTRY = "registry";
+    private static final String FILE_CONFIG_SPLIT_CHAR = ".";
+    private static final String REGISTRY_TYPE = "kubernetes";
+    private static final String NAMESPACE = "namespace";
+
+    private static volatile boolean subscribeListener = false;
+    private static volatile KubernetesRegistryServiceImpl instance;
+    private static final int MAP_INITIAL_CAPACITY = 8;
+    private static  ConcurrentMap<String, List<InetSocketAddress>> clusterAddressMap;
+
+    private static final Configuration FILE_CONFIG = ConfigurationFactory.CURRENT_FILE_INSTANCE;
+
+    private static volatile KubernetesClient kubernetesClient;
+
+    private KubernetesRegistryServiceImpl() {
+    }
+
+
+    static KubernetesRegistryServiceImpl getInstance(KubernetesClient client){
+        if (null == instance) {
+            synchronized (KubernetesRegistryServiceImpl.class) {
+                if (null == instance) {
+                    clusterAddressMap = new ConcurrentHashMap<>(MAP_INITIAL_CAPACITY);
+                    kubernetesClient = client;
+                    instance = new KubernetesRegistryServiceImpl();
+                }
+            }
+        }
+        return instance;
+    }
+
+
+    @Override
+    public void register(InetSocketAddress address) throws Exception {
+        if (!subscribeListener){
+            refreshCluster();
```

Comment:
How do I specify the ip and port I want to register? This information is stored in InetSocketAddress.

Reply:
In the previous PR, I discussed this with @xingfudeshi. There is no need to register in kubernetes. You can get the instance information by running in kubernetes

Reply:
@slievrly I want to resolve this conversation. If there are any questions, please reopen it